### PR TITLE
Add posts on Bevy labels

### PR DESCRIPTION
**Pull request by killercup**

### CODE REVIEW

File: _posts/2022-01-10-bevy-labels.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+---
+title: How Bevy uses Rust traits for labelling
+categories:
+- rust
+- bevy
+---
+Out of curiosity I've recently started following the development of [Bevy],
+a game engine written in [Rust] that.
```
Comment (alice-i-cecile) [inline _posts/2022-01-10-bevy-labels.md]: Sentence fragment

File: _posts/2022-01-10-bevy-labels.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+**Note:** The implementation we arrive at is actually very generic
+-- you can easily apply it to any other Rust project.
+
+[Bevy]: https://bevyengine.org/
+[Rust]: https://www.rust-lang.org/
+
+## How to bevy
+
+Bevy really wants you to use its entity component system architecture
```
Comment (alice-i-cecile) [inline _posts/2022-01-10-bevy-labels.md]:
```suggestion
Bevy really wants you to use its entity-component-system architecture
```

File: _posts/2022-01-10-bevy-labels.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+Imagine this:
+By now we have a whole game built using dozens of systems.
+But for some reason the player movement seems a bit broken,
+like it's rendering one frame too late.
+What is the issue?
+After two hours and too much coffee we realize that
+we wrote `.after("imput")`.
+
+How can we make sure that a simple typo won't break or game again?
```
Comment (alice-i-cecile) [inline _posts/2022-01-10-bevy-labels.md]:
```suggestion
How can we make sure that a simple typo won't break our game again?
```
Oh the irony.

File: _posts/2022-01-10-bevy-labels.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+## Get me out of this stringly-typed mess
+
+So far we've used string to define and refer to labels,
```
Comment (alice-i-cecile) [inline _posts/2022-01-10-bevy-labels.md]:
```suggestion
So far we've used strings to define and refer to labels,
```

File: _posts/2022-01-10-bevy-labels.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+Did you think we were done?
+Oh no! There one more thing:
+`SystemLabel` is not alone!
+There is also `StageLabel`, `AmbiguitySetLabel`, and `RunCriteriaLabel`.
+
+These label types are pretty much all the same,
+but distinct traits in the type system.
+That means you will have to explicitly derive `StageLabel`
+if you want to use your type to refer to a stage;
+you can't use a `SystemLabel` or any other label for that.
```
Comment (alice-i-cecile) [inline _posts/2022-01-10-bevy-labels.md:209]: This is an important point that I think deserves a bit more clarification. The fact that you can't use system labels for stage labels is very helpful: it prevents you from feeding the compiler garbage instructions, and moves those errors from run time to compile time.

Comment (killercup) [reply _posts/2022-01-10-bevy-labels.md:209]: Good point! Will add a paragraph :)

File: _posts/2022-01-10-bevy-labels.md (hunk 1 above)
Comment (killercup) [reply _posts/2022-01-10-bevy-labels.md]: Ah yes, thanks! I moved the ECS description further down

**Pull request merged**